builder/lxc/step_lxc_create.go

```diff
@@ -14,54 +16,90 @@ import (
+		err = fmt.Errorf("Could not read lxc config (%s): %s", config.ConfigFile, err)
+		return "", err
+	}
+	containerConfig.SetRootFs(rootfs)
+	tmpDir, err := ioutil.TempDir("", "lxcconfig")
+	if err != nil {
+		err = fmt.Errorf("Could not create temp directory for lxc config (%s): %s", tmpDir, err)
+		return rootfs, err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = containerConfig.Write(filepath.Join(tmpDir, "lxc.config"))
+	if err != nil {
+		err = fmt.Errorf("Could not write lxc config to %s: %s", filepath.Join(tmpDir, "lxc.config"), err)
+		return rootfs, err
+	}
+
+	commands := make([][]string, 3)
+	commands[0] = []string{"mkdir", containerPath}
+	commands[1] = []string{"tar", "-C", containerPath, "-xf", config.Archive}
+	commands[2] = []string{"cp", filepath.Join(tmpDir, "lxc.config"), filepath.Join(containerPath, "config")}
+
+	err = s.SudoCommands(commands...)
+	return rootfs, err
+}
+
 func (s *stepLxcCreate) Run(state multistep.StateBag) multistep.StepAction {
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
+	errorHandler := func(err error) {
+		state.Put("error", err)
+		ui.Error(err.Error())
+	}
 
-	name := config.ContainerName
-
-	// TODO: read from env
-	lxc_dir := "/var/lib/lxc"
-	rootfs := filepath.Join(lxc_dir, name, "rootfs")
-
-	if config.PackerForce || config.CleanupFirst {
+	if config.PackerForce {
 		s.destroy(config.ContainerName, ui)
 	}
 
-	commands := make([][]string, 3)
-	if config.CloneSource == "" {
-		commands[0] = append(config.EnvVars, []string{"lxc-create", "-n", name, "-t", config.Name, "--"}...)
-		commands[0] = append(commands[0], config.Parameters...)
+	var rootfs string
+	var err error
+	if config.LxcTemplate.Name != "" {
+		ui.Say("Creating container from template...")
+		rootfs, err = s.createFromTemplate(config.ContainerName, config.LxcTemplate)
 	} else {
-		commands[0] = []string{"lxc-clone", "--orig", config.CloneSource, "--new", name}
+		ui.Say(fmt.Sprintf("Creating container from archive: %s", config.RootFs.Archive))
+		rootfs, err = s.createFromRootFs(config.ContainerName, config.RootFs)
 	}
-	// prevent tmp from being cleaned on boot, we put provisioning scripts there
-	// todo: wait for init to finish before moving on to provisioning instead of this
-	commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
-	commands[2] = []string{"lxc-start", "-d", "--name", name}
-
-	ui.Say("Creating container...")
-	for _, command := range commands {
-		log.Printf("Executing sudo command: %#v", command)
-		err := s.SudoCommand(command...)
-		if err != nil {
-			err := fmt.Errorf("Error creating container: %s", err)
-			state.Put("error", err)
-			ui.Error(err.Error())
-			return multistep.ActionHalt
-		}
+	if err != nil {
+		errorHandler(err)
+		return multistep.ActionHalt
+	}
+	ui.Say("Starting container...")
+	if err = s.SudoCommand("lxc-start", "-d", "-n", config.ContainerName); err != nil {
+		errorHandler(fmt.Errorf("Error starting container: %s", err))
```

Comment:
What the. Yeah, you define `errorHandler := func(err error) { ...` but then pass two arguments here. You could probably just break out the function since it's only called twice for two lines but either way why does this line compile? :S